src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -231,6 +231,42 @@ public bool IsSetAsInput
             }
         }
 
+        /// <summary>
+        /// Output nodes are used applications that consume graphs outside of Dynamo such as Optioneering, Optimization, 
+        /// and Dynamo Player. Output nodes can be any node that returns a single output or a dictionary.  
+        /// </summary>
+        [JsonIgnore]
+        public virtual bool IsOutputNode
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        private bool isSetAsOutput = false;
+
+        /// <summary>
+        /// This property is user-controllable via a checkbox and is set to true when a user wishes to include
+        /// this node in the OutputData block of the Dyn file.
+        /// </summary>
+        [JsonIgnore]
+        public bool IsSetAsOutput
```

Comment:
what happens here for CBN node? (the check box is visible, but the user cannot check?)

Reply:
CBN does not have the checkbox visible.  I override `IsOutput = false` the same way we do for `IsInput` and the visibility of the control in xaml is bound to that property

Reply:
can you file a task on CBNs getting figured out?